lib/target.py

```diff
@@ -242,20 +328,30 @@ def __init__(self, triple):
             self.sdk = OSType.Win32
             self.dynamic_library_suffix = ".dll"
             self.executable_suffix = ".exe"
+        elif "darwin" in triple:
+            self.sdk = OSType.MacOSX
+        else:
+            print("Unknown platform")
+
         self.triple = triple
+
         comps = triple.split('-')
-        ArchType.from_string(comps[0])
+        self.arch = ArchType.from_string(comps[0])
 
     @staticmethod
     def default():
-        triple = platform.machine() + "-"
+        arch = ArchType.from_string(platform.machine())
+        triple = ArchType.to_string(arch)
         if platform.system() == "Linux":
-            triple += "linux-gnu"
+            if arch == ArchType.arm:
+                triple += "-linux-gnueabihf"
```

Comment:
See http://clang.llvm.org/docs/CrossCompilation.html

arch = x86, arm, thumb, mips, etc.
sub = for ex. on ARM: v5, v6m, v7a, v7m, etc.
vendor = pc, apple, nvidia, ibm, etc.
sys = none, linux, win32, darwin, cuda, etc.
abi = eabi, gnu, android, macho, elf, etc.

So we probably need another field since triples are not three parts and instead five parts with three reasonably optional fields.
arch[sub][-vendor]-sys[-abi]

However this method is the script's host not the target. So I am not certain which hosts we want to really support here.